examples/multilang/nodejs/mycomponent/index.js

```diff
@@ -0,0 +1,35 @@
+const pulumi = require("@pulumi/pulumi");
+const aws = require("@pulumi/aws");
+
+// TODO: This should be done in the AWS library.  Also, not clear that providers work correctly
+// currently - we appear to end up with class to `Check` a provider before it has been configured -
+// I suspect that the proxy is getting confused as a separate identity even though it's the same
+// URN.
+pulumi.runtime.registerProxyConstructor("pulumi:providers:aws", aws.Provider)
+
+class MyInnerComponent extends pulumi.ComponentResource {
+    constructor(name, args, opts) {
+        super("my:mod:MyInnerComponent", name, {}, opts);
+        this.data = "mydata";
+        this.registerOutputs({
```

Comment:
For the future: any way we can hook the assignment above and automagically track the component's outputs?

Reply:
I think there are two near term options:
1. Rely on registered component outputs
2. Scrape properties off the object using some heuristic

I strongly prefer (1), though it is more of a "change" to what component authors will need to do.  It makes much more clear what the Pulumi Resource Model API of the resource is.